### DISCUSSION TITLE
클로저란 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#23] [변수]
- 클로저란 무엇인가요?

**Comment by publdaze** [comment]
> - 클로저란 무엇인가요?
- 내부 함수가 외부 함수에 있는 스코프에 접근이 가능하여 외부 함수 스코프가 소멸되어도 내부 함수에서 외부 함수의 변수에 접근가능

**Comment by pipisebastian** [comment]
>클로저란 무엇인가요?

- 내부함수가 상위 스코프인 외부함수의 식별자를 참조하고 있을때, 내부함수가 외부함수보다 더 오래 살아있어, 실행이 끝난 외부함수의 상위 스코프(렉시컬 환경)에 접근이 가능한겁니다.
- 내부함수가 외부함수의 렉시컬 환경을 참조하고 있기 때문에, 외부함수가 종료되더라도, 외부함수의 `렉시컬 환경` - `enviroment record` 가 살아있게 됩니다.
